new-new-packages/compiler/src/language_server/semantic_tokens.rs

```diff
@@ -153,42 +154,35 @@ impl<'a> Context<'a> {
-            CstKind::OpeningCurlyBrace { .. } => {}
-            CstKind::ClosingCurlyBrace { .. } => {}
-            CstKind::Arrow { .. } => {}
-            CstKind::Int { .. } => self.add_token(cst.span(), SemanticTokenType::Number),
-            CstKind::Text { .. } => self.add_token(cst.span(), SemanticTokenType::String),
-            CstKind::Identifier { .. } => self.add_token(
-                cst.span(),
-                token_type_for_identifier.unwrap_or(SemanticTokenType::Function),
-            ),
-            CstKind::Symbol { .. } => self.add_token(cst.span(), SemanticTokenType::Symbol),
-            CstKind::LeadingWhitespace { child, .. } => {
-                self.visit_cst(child, token_type_for_identifier)
-            }
-            CstKind::LeadingComment { value, child } => {
-                let span = cst.span();
-                self.add_token(
-                    span.start..span.start + value.len(),
-                    SemanticTokenType::Comment,
-                );
-                self.visit_cst(child, token_type_for_identifier);
+            CstKind::EqualsSign => self.add_token(cst.span.clone(), SemanticTokenType::Operator),
+            CstKind::Comma => {}
+            CstKind::Colon => {}
+            CstKind::OpeningParenthesis => {}
+            CstKind::ClosingParenthesis => {}
+            CstKind::OpeningBracket => {}
+            CstKind::ClosingBracket => {}
+            CstKind::OpeningCurlyBrace => {}
+            CstKind::ClosingCurlyBrace => {}
+            CstKind::Arrow => self.add_token(cst.span.clone(), SemanticTokenType::Operator),
+            CstKind::DoubleQuote => {} // handled by parent
+            CstKind::Octothorpe => {}  // handled by parent
+            CstKind::Whitespace(_) => {}
+            CstKind::Newline(_) => {}
+            CstKind::Comment { octothorpe, .. } => {
+                self.visit_cst(octothorpe, None);
+                self.add_token(cst.span.clone(), SemanticTokenType::Comment);
             }
             CstKind::TrailingWhitespace { child, .. } => {
-                self.visit_cst(child, token_type_for_identifier);
-            }
-            CstKind::TrailingComment { child, value } => {
-                let span = cst.span();
-                self.visit_cst(child, token_type_for_identifier);
-                self.add_token(span.end - value.len()..span.end, SemanticTokenType::Comment);
+                self.visit_cst(child, token_type_for_identifier)
             }
+            CstKind::Identifier { .. } => self.add_token(
+                cst.span.clone(),
+                token_type_for_identifier.unwrap_or(SemanticTokenType::Function),
+            ),
+            CstKind::Symbol { .. } => self.add_token(cst.span.clone(), SemanticTokenType::Symbol),
+            CstKind::Int { .. } => self.add_token(cst.span.clone(), SemanticTokenType::Number),
+            CstKind::Text { .. } => self.add_token(cst.span.clone(), SemanticTokenType::String),
```

Comment:
Now that linebreaks and indentation are not part of the text, maybe we should only report the actual `TextPart`s and quotes?